internal/provider/resource_tfe_workspace_execution_mode.go

```diff
@@ -0,0 +1,179 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+
+package provider
+
+import (
+	"errors"
+	"fmt"
+	"log"
+
+	"github.com/hashicorp/go-tfe"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+)
+
+func resourceTFEWorkspaceExecutionMode() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceTFEWorkspaceExecutionModeCreate,
+		Update: resourceTFEWorkspaceExecutionModeUpdate,
+		Read:   resourceTFEWorkspaceExecutionModeRead,
+		Delete: resourceTFEWorkspaceExecutionModeDelete,
+		Importer: &schema.ResourceImporter{
+			StateContext: schema.ImportStatePassthroughContext,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"agent_pool_id": {
+				Type:     schema.TypeString,
+				Optional: true,
+				ForceNew: true,
```

Comment:
I don't think this ForceNew option is necessary because if execution_mode is "agent", you should be able to update the agent_pool_id on a workspace.